### PR TITLE
Reader: Detect and load platform for twitter, facebook, tumblr

### PR DESCRIPTION
**Pull request by blowery**
This teaches the normalizer and the embed container how to process twitter and facebook "embeds" which are not normal iframe oembeds.

To test, try http://calypso.localhost:3000/read/feeds/25320833

### CODE REVIEW

**Comment by blowery** [comment]
Fixes #746 
Fixes #936

File: client/components/embed-container/index.jsx [hunk 1]
```diff
@@ -57,6 +59,69 @@ function embedInstagram( domNode ) {
...
+}
+
+let fbLoader;
+function embedFacebook( domNode ) {
+	debug( 'processing twitter for', domNode );
+	if ( domNode.hasAttribute( 'data-wpcom-embed-processed' ) ) {
+		return; // already marked for processing
+	}
+
+	domNode.setAttribute( 'data-wpcom-embed-processed', '1' );
```
Comment (TooTallNate) [inline client/components/embed-container/index.jsx]: I wonder if any of the browsers that we support at this point don't support the [`dataset` API](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset)?

Comment (blowery) [inline client/components/embed-container/index.jsx]: they do, but it's "slower". see https://bugzilla.mozilla.org/show_bug.cgi?id=802157

**Comment by blowery** [comment]
Added support for tumblr and fixed up some bits in the facebook embedder.

**Comment by blowery** [comment]
Fixes #3939

**Comment by TooTallNate** [comment]
Nice DRYing it up! LGTM :shipit:

**Pull request merged**